src/python/pants/init/logging.py

```diff
@@ -65,8 +65,12 @@ def __repr__(self) -> str:
 class ExceptionFormatter(Formatter):
     """Uses the `--print-exception-stacktrace` option to decide whether to render stacktraces."""
```

Comment:
It looks like there's been a lot of work in `init/logging.py` on things that used to be in `ExceptionSink`. The only reason `ExceptionSink` was created was to centralize exception handling and formatting: see #6530. At this point I would actually advocate for potentially removing `ExceptionSink` entirely along with the skipped tests, as the current situation actually seems actively harmful and counter to the point of the `ExceptionSink` class in the first place (I can do that in a followup).

I have no clue how to avoid this happening in the future, but things like the backtrace printing twice as you mentioned on slack is exactly what I would expect if the logic was essentially duplicated elsewhere in pants. It feels like there's been some unnecessary code siloing going on that has led to directly recreating some of the same very tricky issues with error handling (because the point of `ExceptionSink` was to centralize error handling to fight back at exactly these issues).

In the future I would absolutely feel empowered to make these changes, and I think I would specifically recommend keeping an eye out for error handling in too many places if you ever feel like a refactoring.

Reply:
@cosmicexplorer yeah it looks like a lot of the code in `ExceptionSink` was written before the rust engine, so there's now things that code in the rust engine and/or Python code that closely interacts with the rust engine (e.g. `init/logging.py`) handles now. It might make sense to remove `ExceptionSink` entirely, but I think if we just deleted the code now it would at least break tests that expect it, and possibly break functionality we do care about keeping. Also `ExceptionSink` concerns around writing to log files in `.pids` seem relevant to e.g. https://github.com/pantsbuild/pants/issues/3511

I actually think the double backtrace thing is *not* related to any ExceptionSink code, though. I'm debugging that right now and it seems like it has to do with the Engine backtrace generated by Rust engine error handling attaching the same string to more than one Python exception. Still investigating though.